common/v2/services/Store/StoreProvider.tsx

```diff
@@ -59,6 +70,67 @@ export const StoreProvider = ({ children }: { children: React.ReactNode }) => {
     [accounts]
   );
 
+  useEffect(() => {
+    setPendingTransactions(
+      accounts
+        .map(account =>
+          account.transactions.map(tx => {
+            return { ...tx, network: account.network };
+          })
+        )
+        .flatMap(tx => tx)
+        .filter((txObject: ITxReceipt) => txObject.stage === ITxStatus.PENDING) || []
+    );
+  }, [accounts]);
+
+  // A pending transaction is detected.
+  useEffect(() => {
+    if (pendingTransactions.length <= 0) {
+      return;
+    }
+    // This interval is used to poll for status of txs.
+    const blockNumInterval = setInterval(() => {
+      pendingTransactions.forEach((pendingTransactionObj: any) => {
+        const network: Network = pendingTransactionObj.network;
+        if (!network) {
+          return;
+        }
+        const provider = new ProviderHandler(network);
+
+        provider.getTransactionByHash(pendingTransactionObj.hash).then(transactionReceipt => {
+          if (!transactionReceipt) {
+            return;
+          }
+          const receipt = fromTxReceiptObj(transactionReceipt);
+          if (!receipt) {
+            return;
+          }
```

Comment:
Why would `fromTxReceiptObj` fail?
I thought it was a pure function behaving as a transducer

Reply:
If its unable to fetch the network given tx chain id, then it returns undefined.